Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/BackgroundTask/AddIshBackgroundTask.cs

```diff
@@ -0,0 +1,214 @@
+    /// <para type="description">ParameterGroup uses BackgroundTask25 API to create BackgroundTask.</para>
+    /// <para type="description">IshObjectsGroup uses content object(s) passed via pipeline or as a parameter to trigger BackgroundTask event using DocumentObj25.RaiseEventByIshLngRefs.</para>
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// New-IshSession -WsBaseUrl "https://example.com/ISHWS/" -PSCredential "Admin"
+    /// $ishBackgroundTask = Get-IshFolderContent -FolderPath "General\MyFolder" -VersionFilter Latest -Recurse | Add-IshBackgroundTask -EventType "SMARTTAG"
+    /// </code>
+    /// <para>Add BackgroundTask with event type"SMARTTAG" for the objects located under the "General\MyFolder" path</para> 
+    /// </example>
+    /// <example>
+    /// <code>
+    /// New-IshSession -WsBaseUrl "https://example.com/ISHWS/" -PSCredential "Admin"
+    /// $ishBackgroundTask  = Add-IshBackgroundTask -EventType "PUSHTRANSLATIONS" -EventDescription "Custom event description" -RawInputData $rawData
+    /// </code>
+    /// <para>Add background task with the event type "PUSHTRANSLATIONS" and provided event description and input data</para> 
+    /// </example>
+
+    [Cmdlet(VerbsCommon.Add, "IshBackgroundTask", SupportsShouldProcess = false)]
+    [OutputType(typeof(IshBackgroundTask))]
+    public sealed class AddIshBackgroundTask : BackgroundTaskCmdlet
+    {
+        /// <summary>
+        /// <para type="description">The IshSession variable holds the authentication and contract information. This object can be initialized using the New-IshSession cmdlet.</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParameterGroup")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshObjectsGroup")]
+        [ValidateNotNullOrEmpty]
+        public IshSession IshSession { get; set; }
+
+        /// <summary>
+        /// <para type="description">Type of the event (e.g. SMARTTAG)</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParameterGroup")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = false, ParameterSetName = "IshObjectsGroup")]
+        [ValidateNotNullOrEmpty]
+        public string EventType { get; set; }
+
+        /// <summary>
+        /// <para type="description">The input data for the background task.</para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParameterGroup")]
+        [ValidateNotNullOrEmpty]
+        public byte[] RawInputData { get; set; }
+
+        /// <summary>
+        /// <para type="description">Description of the event</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParameterGroup")]
```

Comment:
Mandatory string -EventDescription (use in EventMonitor25.StartEvent with maxProgress = 100 and generated progressId/eventId is passsed to CreateBackgroundTask) ... or was it implemented differently?